mccn/mccn.py

```diff
@@ -71,13 +83,33 @@ def plot(xx: xarray.Dataset):
 
     def load(self, col_id: str, bands: Optional[List[str]] = None, groupby: str = "id",
              crs: Optional[str] = None, geobox: Optional[GeoBox] = None, lazy: bool = False,
-             source: Optional[Dict[str, str]] = None):
-        xx = self.load_stac(col_id, bands=bands, groupby=groupby, crs=crs, geobox=geobox,
-                            lazy=lazy)
-
+             source: Optional[Dict[str, str]] = None, mask=None):
+        """
+        Load the STAC items for a given collection ID into an xarray dataset. Several options are
+        available for sub-selection and transformation of the data upon loading.
+        :param col_id: STAC collection id for study.
+        :param bands: List of band names to load, defaults to All. Also accepts single band name as
+        input
+        :param groupby:
+        :param crs:
+        :param geobox:
+        :param lazy:
+        :param source:
+        :param mask:
+        :return:
+        """
+        xx: Dataset | DataArray = self._load_stac(col_id, bands=bands, groupby=groupby, crs=crs,
+                                                  geobox=geobox, lazy=lazy)
+        if mask and geobox:
+            vector_array = stac_load_vector(
+                list(self._query(col_id)), gbox=geobox
+            )
+            # TODO: Combine rasterised vector layers with other raster layers. Below is placeholder
+            # TODO: to appease ruff.
+            xx = xarray.combine_by_coords([xx, vector_array])
```

Comment:
I suppose this will not work well if you have dataset with different `dim`. Let's have two more variables at init for `X_name` and `Y_name` and rename the X and Y dimensions to them after spatial query/combine.

Reply:
Dont focus too much on combination, this is a temporary fix/prototype so the code works. We will be working on it properly this sprint.